### PR TITLE
feat: [CDS-40821]: added gitOps icon for gitOps services

### PR DESCRIPTION
**Pull request by Sagar-Rajput7**
#### Screenshots

https://user-images.githubusercontent.com/98325173/180851795-44e82246-6165-43e1-ab4e-9a6041961e99.mov

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by vikrant-harness — APPROVED** [review]

**Review by AshwinBhatkal-Harness — DISMISSED** [review]

File: src/modules/70-pipeline/utils/DeploymentTypeUtils.ts [hunk 1]
```diff
@@ -23,5 +23,6 @@ export const deploymentTypeIcon: Record<string, IconName> = {
   NativeHelm: 'service-helm',
   ServerlessAwsLambda: 'service-serverless',
   Ssh: 'secret-ssh',
-  WinRm: 'command-winrm'
+  WinRm: 'command-winrm',
+  'Kubernetes:gitOps': 'gitops-green' //type specific to gitops
```
Comment (bhavya-sinha) [inline src/modules/70-pipeline/utils/DeploymentTypeUtils.ts:27]: Why is deployment type like this. Moreover the type should not be any random string. It should be one of deployment type

Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/utils/DeploymentTypeUtils.ts:27]: This is for gitops services (currently we support gitops only with K8s), 
BE is sending this in the response here - the format here is "DeploymentType:gitOps"

Comment (kapilkumawat86) [reply src/modules/70-pipeline/utils/DeploymentTypeUtils.ts:30]: Can we just use KubernetesGitOps

Comment (bhavya-sinha) [reply src/modules/70-pipeline/utils/DeploymentTypeUtils.ts:30]: Sounds better. @Sagar-Rajput7

Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/utils/DeploymentTypeUtils.ts:30]: BE is merged for this so I will tell BE to change this then.

**Review by bhavya-sinha — APPROVED** [review]

**Review by vikrant-harness — APPROVED** [review]

File: src/modules/70-pipeline/utils/DeploymentTypeUtils.ts [hunk 2]
```diff
@@ -9,6 +9,8 @@ import type { IconName } from '@harness/uicore'
 import type { ServiceDefinition } from 'services/cd-ng'
 import type { StringKeys } from 'framework/strings'
 
+export type ServiceTypes = ServiceDefinition['type'] & 'KubernetesGitOps'
```
Comment (AshwinBhatkal-Harness) [inline src/modules/70-pipeline/utils/DeploymentTypeUtils.ts:12]: Will backend be adding this type - KubernetesGitOps - later?

**Review by AshwinBhatkal-Harness — APPROVED** [review]

File: src/modules/70-pipeline/utils/DeploymentTypeUtils.ts (hunk 2 above)
Comment (Sagar-Rajput7) [reply src/modules/70-pipeline/utils/DeploymentTypeUtils.ts:12]: yeah i have asked them and shared this String with them

**Pull request merged**